code/datums/station_traits/job_traits.dm

```diff
@@ -93,3 +108,55 @@
 	// monkey carries the crates, the age of robot is over
 	if(GLOB.cargo_ripley)
 		qdel(GLOB.cargo_ripley)
+
+/datum/station_trait/job/bridge_assistant
+	name = "Bridge Assistant"
+	button_desc = "Sign up to become the Bridge Assistant and watch over the Bridge."
+	weight = 2
+	report_message = "We have installed a Bridge Assistant on your station."
+	show_in_report = TRUE
+	can_roll_antag = CAN_ROLL_PROTECTED
+	job_to_add = /datum/job/bridge_assistant
+
+/datum/station_trait/job/bridge_assistant/New()
+	. = ..()
+	RegisterSignal(SSatoms, COMSIG_SUBSYSTEM_POST_INITIALIZE, PROC_REF(add_coffeemaker))
+
+/datum/station_trait/job/bridge_assistant/on_lobby_button_update_overlays(atom/movable/screen/lobby/button/sign_up/lobby_button, list/overlays)
+	. = ..()
+	overlays += "bridge_assistant"
+
+/// Creates a coffeemaker in the bridge, if we don't have one yet.
+/datum/station_trait/job/bridge_assistant/proc/add_coffeemaker(datum/source)
+	SIGNAL_HANDLER
+	var/area/bridge = GLOB.areas_by_type[/area/station/command/bridge]
+	if(isnull(bridge)) //no bridge, what will he assist?
+		return
+	var/list/possible_coffeemaker_positions = list(/area/station/command/bridge, /area/station/command/meeting_room)
+	for(var/possible_position in possible_coffeemaker_positions) //don't spawn a coffeemaker if there is already one on the bridge
+		var/area/possible_area = GLOB.areas_by_type[possible_position]
+		if(possible_area && (locate(/obj/machinery/coffeemaker) in possible_area))
+			return
+	var/list/tables = list()
+	for(var/obj/structure/table/table in bridge)
+		var/turf/table_turf = get_turf(table)
+		if(table_turf.is_blocked_turf(ignore_atoms = list(table))) //don't spawn a coffeemaker on a fax machine or smth
+			continue
+		tables += table
+	if(!length(tables))
+		return
+	var/picked_table = pick(tables)
+	tables -= picked_table
+	var/picked_turf = get_turf(picked_table)
+	if(length(tables))
+		var/another_table = pick(tables)
+		for(var/obj/item/thing_on_table in picked_turf) //if there's paper bins or other shit on the table, get that off
+			if(thing_on_table == picked_table)
+				continue
+			thing_on_table.forceMove(get_turf(another_table))
```

Comment:
You account for paper bins, but you don't account for recharges or cell chargers.

You will also need a further check to prevent this loop from stealing wall mounts. It will take intercoms, for example. 

You also check that `thing != picked_table`, but the loop is cast to `/obj/item`, so the check is redundant. Though if you cast it to `/obj` be careful not to grab dir-windows or other wallmounts.